functorch/experimental/cond.py

```diff
@@ -151,6 +154,22 @@ def cond_python_dispatcher(*args):
     return cond(*args)
 
 
+@cond.py_impl(torch._C._functorch.TransformType.Functionalize)
+def cond_functionalize(interpreter, pred, true_fn, false_fn, inputs):
+    reapply_views = interpreter.functionalizeAddBackViews()
+    mode = 'mutations_and_views' if reapply_views else 'mutations'
+    unwrapped_inputs = _unwrap_all_tensors_from_functional(inputs, reapply_views=reapply_views)
+    functional_true_fn = make_fx(functionalize(true_fn, remove=mode))(*unwrapped_inputs)
+    functional_false_fn = make_fx(functionalize(false_fn, remove=mode))(*unwrapped_inputs)
+    _ = ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.FuncTorchDynamicLayerFrontMode))
+    return _wrap_all_tensors_to_functional(cond(pred, functional_true_fn, functional_false_fn, unwrapped_inputs), level=interpreter.level())
```

Comment:
The right way to "dispatch past the functionalization transform" is to use interpreter.lower(), instead of excluding FuncTorchDynamicLayer* keys:
```
with interpreter.lower():
    return _wrap_all_tensors_to_functional(cond(pred, functional_true_fn, functional_false_fn, unwrapped_inputs), level=interpreter.level())
```